engine/execution/provider/engine.go

```diff
@@ -346,36 +341,6 @@ func (e *Engine) deliverChunkDataResponse(chunkDataPack *flow.ChunkDataPack, req
 	lg.Info().Msg("chunk data pack request successfully replied")
 }
 
-func (e *Engine) ensureAuthorized(chunkID flow.Identifier, originID flow.Identifier) (*flow.Identity, error) {
-	blockID, err := e.execState.GetBlockIDByChunkID(chunkID)
-	if err != nil {
-		return nil, engine.NewInvalidInputErrorf("cannot find blockID corresponding to chunk data pack: %w", err)
-	}
-
-	authorizedAt, err := e.checkAuthorizedAtBlock(blockID)
-	if err != nil {
-		return nil, engine.NewInvalidInputErrorf("cannot check block staking status: %w", err)
-	}
-	if !authorizedAt {
-		return nil, engine.NewInvalidInputErrorf("this node is not authorized at the block (%s) corresponding to chunk data pack (%s)", blockID.String(), chunkID.String())
-	}
-
-	origin, err := e.state.AtBlockID(blockID).Identity(originID)
-	if err != nil {
-		return nil, engine.NewInvalidInputErrorf("invalid origin id (%s): %w", origin, err)
-	}
-
-	// only verifier nodes are allowed to request chunk data packs
-	if origin.Role != flow.RoleVerification {
```

Comment:
We checked if the requester is a VN. Should we add this check back?

Reply:
We do check if the request is from a VN or EN here https://github.com/onflow/flow-go/blob/master/network/channels/channels.go#L210